File: NET/JNetCall.Sharp.StdIO/Client/StdIOClient.cs
using System.IO;
using JNetBase.Sharp.Threads;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Client.Tools;
using JNetCall.Sharp.Impl.Enc;
using JNetCall.Sharp.Impl.IO;

namespace JNetCall.Sharp.Client
{
    // ReSharper disable once InconsistentNaming
    public static class StdIOClient
    {
        public static T Create<T>(string exe) where T : class
        {
            var pool = new ThreadExecutor();
            var protocol = new JarTransport(exe, InitDefault);
            var handler = new ClassProxy(protocol, pool);
            handler.Listen();
            return ClientHelper.Create<T>(handler);
        }

        private static ISendTransport InitDefault(Stream stdIn, Stream stdOut)
        {
            var enc = new BinaryEncoding();
            return new StreamTransport(enc, stdIn, stdOut);
        }
    }
}

File: JVM/jnetcall-java-stdio/src/main/java/jnetcall/java/client/StdIOClient.java
package jnetcall.java.client;

import java.io.InputStream;
import java.io.OutputStream;

import jnetbase.java.threads.ThreadExecutor;
import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.client.tools.ClientHelper;
import jnetcall.java.impl.enc.BinaryEncoding;
import jnetcall.java.impl.io.StreamTransport;

public final class StdIOClient {

    public static <T> T create(Class<T> clazz, String exe) {
        ThreadExecutor pool = new ThreadExecutor();
        ExeTransport protocol = new ExeTransport(exe, StdIOClient::initDefault);
        ClassProxy handler = new ClassProxy(protocol, pool);
        handler.listen();
        return ClientHelper.create(clazz, handler);
    }

    private static ISendTransport initDefault(InputStream stdIn, OutputStream stdOut) {
        BinaryEncoding enc = new BinaryEncoding();
        return new StreamTransport(enc, stdIn, stdOut);
    }
}
